server/internal/core/application/covenantless.go

```diff
@@ -528,14 +538,20 @@ func (s *covenantlessService) SpendNotes(ctx context.Context, notes []note.Note)
 		return "", fmt.Errorf("failed to create tx request: %s", err)
 	}
 
+	// add the server pubkey as a signer
+	signerPubkeys = append(signerPubkeys, hex.EncodeToString(s.serverSigningPubKey.SerializeCompressed()))
+
+	request.AddSignerPubKeys(signerPubkeys)
+	request.AddSigningType(signingType)
+
```

Comment:
why do we do this here? the server key could be added only once, when the requests are popped from the queue and the batching is processed.

This way instead we would end up duplicating the server key in memory for every note redeemed in the same round, wouldn't we?